libs/test_positivity.py

```diff
@@ -21,35 +22,83 @@
 _log = structlog.get_logger()
 
 
-@dataclasses.dataclass
-class Method:
+class Method(ABC):
     """A method of calculating test positivity"""
 
-    name: str
-    numerator: FieldName
-    denominator: FieldName
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        pass
 
     @property
+    @abstractmethod
     def columns(self) -> Set[FieldName]:
-        return {self.numerator, self.denominator}
+        pass
 
-    def calculate(self, delta_df: pd.DataFrame) -> pd.DataFrame:
+    @abstractmethod
+    def calculate(self, df: pd.DataFrame, delta_df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
We now pass in the "normal" dataframe as well as the "delta" dataframe, for the pass-thru method.